### PR TITLE
Update github New Issue workflow to mimic Remix

### PR DESCRIPTION
**Pull request by brophdawg11**
* Update the `.github/ISSUE_TEMPLATE/config.yml` configuration to send feature requests through the new Proposal workflow
* Added a new template for documentation issues to add a `[Docs]:` prefix and the `docs` label.  
* Also did some cleanup of the `examples/data-router` app since we link to that from the issue template now

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 1]
```diff
@@ -11,18 +11,21 @@ body:
...
-        you ask a question, here are some resources to get help first:
+        The issue tracker is meant for bug reports only. This isn't the best
+        place for support or usage questions. Questions here don't have as much
+        visibility as they do elsewhere. Before you ask a question, here are
+        some resources to get help first:
 
         - Read the docs: https://reactrouter.com
         - Check out the list of frequently asked questions: https://reactrouter.com/start/faq
         - Explore examples: https://reactrouter.com/start/examples
+        - Ask in chat: https://rmx.as/discord
```
Comment (brophdawg11) [inline .github/ISSUE_TEMPLATE/bug_report.yml:22]: Removed the old Reactiflux discord link in favor of the Remix discord, and moved it above stack overflow

File: .github/ISSUE_TEMPLATE/bug_report.yml [hunk 2]
```diff
@@ -11,18 +11,21 @@ body:
...
         - Look for/ask questions on Stack Overflow: https://stackoverflow.com/questions/tagged/react-router
-        - Ask in chat: https://discord.gg/6RyV8n8yyM
 
-        ### Test Case Starter:
-        https://stackblitz.com/github/remix-run/react-router/tree/main/examples/basic?file=src/App.tsx
+        ### Test Case Starters:
+
+        * (Using `<RouterProvider>`)[https://stackblitz.com/github/remix-run/react-router/tree/main/examples/data-router?file=src/App.tsx]
+        * (Using `<BrowserRouter>`)[https://stackblitz.com/github/remix-run/react-router/tree/main/examples/basic?file=src/App.tsx]
```
Comment (brophdawg11) [inline .github/ISSUE_TEMPLATE/bug_report.yml:28]: Link to separate examples of `RouterProvider`/`BrowserRouter` for clarity

File: .github/ISSUE_TEMPLATE/config.yml [hunk 3]
```diff
@@ -1,5 +1,20 @@
...
-    about: This is a bug tracker, not a support system. For usage questions, please use Stack Overflow where there are a lot more people ready to help you out. Thanks!
+    about: Open a question in Stack Overflow with the react-router label
+  - name: 🤔 Usage Question (Github Discussions)
+    url: https://github.com/remix-run/remix/discussions/new?category=q-a
+    about: Open a Discussion in GitHub wih the `Q&A` label
+  - name: 💬 Remix Discord Channel
+    url: https://rmx.as/discord
+    about: Interact with other people using React Router and Remix 📀.  There's
+      plenty of channels for general discussions as well as a threaded `#help`
+      channel in here where you can ask for help with you issue.
```
Comment (brophdawg11) [inline .github/ISSUE_TEMPLATE/config.yml:20]: Mimic https://github.com/remix-run/remix/issues/new/choose from Remix

Screenshot 2023-01-09 at 2 38 25 PM

File: examples/data-router/src/app.tsx [hunk 4]
```diff
@@ -1,50 +1,35 @@
...
-  DeferredChild,
-  todosAction,
-  todosLoader,
-  TodosList,
-  TodosBoundary,
-  todoLoader,
-  Todo,
-  sleep,
-  AwaitPage,
-} from "./routes";
```
Comment (brophdawg11) [inline examples/data-router/src/app.tsx:27]: Got rid of the separate `routes.tsx` file so everything is in `app.tsx` now

File: .github/ISSUE_TEMPLATE/documentation_isse.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
+name: 📚 Documentation Issue
```
Comment (brophdawg11) [inline .github/ISSUE_TEMPLATE/documentation_isse.yml:1]: We have a lot of documentation updates filed as bugs so added a separate template here for those

**Review by chaance — APPROVED** [review]

**Review by jacob-ebey — APPROVED** [review]

**Pull request merged**